test/units/ansible_test/test_docker_util.py

```diff
@@ -0,0 +1,131 @@
+""".lstrip()  # noqa: E501
+
+PODMAN_OUTPUT = """
+[
+    {
+        "id": "dd3d10e03dd3580de865560c3440c812a33fd7a1fca8ed8e4a1219ff3d809e3a",
+        "names": [
+            "quay.io/ansible/centos7-test-container:1.8.0"
+        ],
+        "digest": "sha256:6e5d9c99aa558779715a80715e5cf0c227a4b59d95e6803c148290c5d0d9d352",
+        "created": "2019-04-02T00:59:39.234584184Z",
+        "size": 702761933
+    },
+    {
+        "id": "b0f914b26cc1088ab8705413c2f2cf247306ceeea51260d64c26894190d188bd",
+        "names": [
+            "quay.io/ansible/centos7-test-container:latest"
+        ],
+        "digest": "sha256:d8431aa74f60f4ff0f1bd36bc9a227bbb2066330acd8bf25e29d8614ee99e39c",
+        "created": "2020-06-11T22:05:58.382459136Z",
+        "size": 578513505
+    }
+]
+""".lstrip()
+
+
+@pytest.fixture
+def docker_images():
+    from ansible_test._internal.docker_util import docker_images
+    return docker_images
+
+
+@pytest.fixture
+def ansible_test(ansible_test):
+    import ansible_test
+    return ansible_test
+
+
+@pytest.fixture
+def subprocess_error():
+    from ansible_test._internal.util import SubprocessError
+    return SubprocessError
+
+
+@pytest.mark.parametrize(
+    ('returned_items_count', 'patched_dc_stdout'),
+    (
+        (3, (DOCKER_OUTPUT_MULTIPLE, '')),
+        (2, (PODMAN_OUTPUT, '')),
+        (0, ('', '')),
+    ),
+    ids=('docker JSONL', 'podman JSON sequence', 'empty output'))
+def test_docker_images(docker_images, mocker, returned_items_count, patched_dc_stdout):
+    mocker.patch(
+        'ansible_test._internal.docker_util.docker_command',
+        return_value=patched_dc_stdout)
+    ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+    assert len(ret) == returned_items_count
+
+
+def test_podman_fallback(ansible_test, docker_images, subprocess_error, mocker):
+    '''Test podman >2 && <2.2 fallback'''
+
+    cmd = ['docker', 'images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}']
+    docker_command_results = [
+        subprocess_error(cmd, status=1, stderr='function "json" not defined'),
+        (PODMAN_OUTPUT, ''),
+    ]
+    mocker.patch(
+        'ansible_test._internal.docker_util.docker_command',
+        side_effect=docker_command_results)
+
+    ret = docker_images('', 'quay.io/ansible/centos7-test-container')
+    calls = [
+        call(
+            '',
+            ['images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}'],
+            capture=True,
+            always=True),
+        call(
+            '',
+            ['images', 'quay.io/ansible/centos7-test-container', '--format', 'json'],
+            capture=True,
+            always=True),
+    ]
+    ansible_test._internal.docker_util.docker_command.assert_has_calls(calls)
+    assert len(ret) == 2
```

Comment:
I guess more of the preparation activities could be moved outside making the tests cleaner:
```suggestion
DOCKER_IMAGES_CALLS = [
    call(
        '',
        ['images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}'],
        capture=True,
        always=True),
    call(
        '',
        ['images', 'quay.io/ansible/centos7-test-container', '--format', 'json'],
        capture=True,
        always=True),
]

@pytest.fixture
def docker_command_patch_kwargs(docker_images, mocker, request):
    patch_kwargs = request.param
    if not patch_kwargs:
        return
    mocker.patch(
        'ansible_test._internal.docker_util.docker_command',
        **patch_kwargs
    )


@pytest.mark.parametrize(
    ('returned_items_count', 'patched_dc_stdout', dc_calls_num),
    (
        (3, {'return_value': (DOCKER_OUTPUT_MULTIPLE, '')}),
        (2, {'return_value': (PODMAN_OUTPUT, '')}),
        (0, {'return_value': ('', '')}),
    ),
    indirect=('ansible_module_args', ),
    ids=('docker JSONL', 'podman JSON sequence', 'empty output'),
)
@pytest.mark.usefixtures('docker_command_patch_kwargs')
def test_docker_images(returned_items_count, patched_dc_stdout, dc_calls_num, ansible_test):
    ret = docker_images('', 'quay.io/ansible/centos7-test-container')

    assert len(ret) == returned_items_count

    ansible_test._internal.docker_util.docker_command.assert_has_calls(
        DOCKER_IMAGES_CALLS[:1],
    )


def test_podman_fallback(ansible_test, docker_images, subprocess_error, mocker):
    '''Test podman >2 && <2.2 fallback'''

    cmd = ['docker', 'images', 'quay.io/ansible/centos7-test-container', '--format', '{{json .}}']
    docker_command_results = [
        subprocess_error(cmd, status=1, stderr='function "json" not defined'),
        (PODMAN_OUTPUT, ''),
    ]
    mocker.patch(
        'ansible_test._internal.docker_util.docker_command',
        side_effect=docker_command_results)

    ret = docker_images('', 'quay.io/ansible/centos7-test-container')
    ansible_test._internal.docker_util.docker_command.assert_has_calls(DOCKER_IMAGES_CALLS)
    assert len(ret) == 2
```